code/krepel/container/dynamic_array.d

```diff
@@ -0,0 +1,148 @@
+module krepel.container.dynamic_array;
+
+import krepel.memory;
+import krepel.algorithm : Max;
+import Meta = krepel.meta;
+
+struct DynamicArray(T, A = typeof(null))
+{
+  enum size_t MinimumElementAllocationCount = 16;
+
+  alias ElementType = T;
+
+  static if(is(A == typeof(null)))
+  {
+    alias AllocatorType = typeof(GlobalAllocator);
+    AllocatorType* AllocatorPtr = &GlobalAllocator;
+  }
+  else
+  {
+    alias AllocatorType = A;
+    AllocatorType* AllocatorPtr;
+  }
+
+  ElementType[] Capacity;
```

Comment:
I agree. The dynamic array implementation was just quickly hacked together to have something to use an allocator with :)